### DISCUSSION TITLE
Updating the coordinates / text not working.

### DISCUSSION THREAD

**Discussion by harsimarriar96** [fabricjs/fabric.js#8687] [Q&A]
Hi,

I have been currently trying to update the text and coordinates of a fabricjs object by setting the canvas._objects[0].set({text: 'xyz', dirty: true}) but it is not updating.

I have tried without making it dirty explicitly and tried canvas.renderAll() as well. Seems like a strange behavior. 

If I check the the actual assigned variable, the values are updated there but the updates are not showing on the canvas.

Any suggestions would be appreciated.

**Comment by ShaMan123** [comment]
Seems you are confused.
`dirty` controls rendering state, `setCoords` controls coords state.
http://fabricjs.com/fabric-gotchas

**Reply by harsimarriar96** [reply]
I understand that we need to setCoords() whenever updating the coordinates, but my primary question is that whatever I am changing for a particular object, it's not rendering on the screen even while using .setCoords or updating text or by changing dirty controls manually as suggested in other posts on stackoverflow.
